cmd/cluster-agent/admission/server.go

```diff
@@ -16,29 +16,55 @@ import (
 	"net/http"
 	"time"
 
-	"github.com/DataDog/datadog-agent/pkg/clusteragent/admission"
 	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver/common"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/certificate"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 
 	admiv1beta1 "k8s.io/api/admission/v1beta1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/kubernetes"
 )
 
 const jsonContentType = "application/json"
 
-// RunServer creates and start a k8s admission webhook server
-func RunServer(mainCtx context.Context) error {
-	mux := http.NewServeMux()
-	mux.HandleFunc("/mutate", mutateHandler)
+type admissionFunc func(*admiv1beta1.AdmissionRequest) (*admiv1beta1.AdmissionResponse, error)
+
+type Server struct {
+	mux *http.ServeMux
+}
+
+// NewServer creates an admission webhook server.
+func NewServer() *Server {
+	return &Server{
+		mux: http.NewServeMux(),
+	}
+}
+
+// Register adds an admission webhook handler.
+// Register must be called to register the desired webhook handlers before calling Run.
+func (s *Server) Register(uri string, f admissionFunc) {
+	s.mux.HandleFunc(uri, func(w http.ResponseWriter, r *http.Request) {
+		mutateHandler(w, r, f)
+	})
+}
+
+// Run starts the kubernetes admission webhook server.
+func (s *Server) Run(mainCtx context.Context, client kubernetes.Interface) error {
 	server := &http.Server{
 		Addr:    fmt.Sprintf(":%d", config.Datadog.GetInt("admission_controller.port")),
-		Handler: mux,
+		Handler: s.mux,
 		TLSConfig: &tls.Config{
 			GetCertificate: func(info *tls.ClientHelloInfo) (*tls.Certificate, error) {
-				// TODO: implement me
-				return &tls.Certificate{}, nil
+				secretNs := common.GetResourcesNamespace()
+				secretName := config.Datadog.GetString("admission_controller.certificate.secret_name")
+				cert, err := certificate.GetCertificateFromSecret(secretNs, secretName, client)
+				if err != nil {
+					log.Errorf("Couldn't fetch certificate: %v", err)
+				}
+				return cert, nil
```

Comment:
nits: extra this function to be able to unit-test it

Reply:
yes, will do when we build the cert file support next 👍